### PR TITLE
Milestone Five: DOM

### PR DESCRIPTION
**Pull request by phylischumba**
### In this Milestone, I did:

- Added DOM manipulation to the calculator buttons
- Fix linter errors

### CODE REVIEW

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Change required❌

# 🙋‍♂️ @Phylis 

#### Good job done so far, here are a few issues we have to address to get this project approved🤗.

* [x] Currently the calculator throws an error when the number is divided by zero, ideally we should expect `undefined`.

* [x] Currently the calculator returns a wrong response when we try to compute the percentage of a number ie `num1%`. 

* [x] Currently the calculator returns a wrong response when we perform this computation `num1+num2` then select `%`, ideally we should expect `(num1+num2)/100`.

### That is it kindly make the change a submit a re-review🙂.

#### Happy coding👨‍💻

Addo Forison

**Comment by phylischumba** [comment]
Hey @Forison, thanks for the review. Looked in the changes and have worked on them

File: src/components/App.js [hunk 1]
```diff
@@ -4,11 +4,30 @@ import ButtonPanel from './ButtonPanel';
...
+  handleClick(buttonName) {
+    const calcData = { ...this.state };
+    this.setState(calculate(calcData, buttonName));
+  }
+
+  render() {
+    const { next, result, operation } = this.state;
+    return (
+      <div className="app">
+        <Display result={next || result || operation || '0'} />
```
Comment (uryelah) [inline src/components/App.js]: - [x] You should pass all the three, or at least next and result, props to the Display to make this work like a regular calculator application.
Open the calculator app of your machine or cellphone and notice how it always shows the operation and the next numbers you click.
For example, if a user click on this sequence of buttons: '5', '+', '2'
The Display should show:
`5 + 2`
And when they click '=' it should show either:
```
5 + 2
      7
```
With the history of what was calculated at the top and at the bottom the result, or just the result:
`7`

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Phylis Chepchumba**, I'm Sarah your reviewer for this PR, sorry for the wait 🐢  👩‍🚀 

Good job fixing the division by zero issues. 
There are a couple more issues with the core functionality of your calculator but after fixing those you'll be good to go.

Rock on! 🎸 👨‍🎤 

# Required Changes 🏗️ 🐛 💡

- [x] Make sure to keep the deployed link updated, you should always deploy again after making changes to the app.

(Check the other change below in the comments)

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by phylischumba** [comment]
Hey @uryelah, thanks for your feedback. I was able to implement a part of it but the other became a bit challenging

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Changes Required❌.
#  🙋‍♂️ @Phylis05 
#### Good job on improving the user experience of the application, however there are issues with the functionality of the application🤗.
* [x] On select `-/+` the current state of the operands should be changed such that if they are positive they should become negative, vice versa. Currently, it only affects the second operand(next) without that affecting the first(total).
* [x] The default zero value should be replaced by the first number selected, for instance instead of `07` We have `7`.
* [x] On selecting percentage, should trigger a response in decimal of the number that we ought to find the percentage for. for instance, `20%` should return `0.20`, that way we can do further computations with it.
* > Kindly, watch the [video](https://www.loom.com/share/8ebb629471874152a161672e44a424ed) for clarifications.
### That is it kindly make the change and submit a re-review👨‍🔧.
#### Happy coding👨‍💻
Addo Forison

**Review by Torres-ssf — APPROVED** [review]
### Milestone approved :+1: :clap: 
Hi there @Phylis05 :hand: 
Great job implementing the requested changes.
Your project it's approved. 
You can merge the PR now.

Happy Coding :computer: 
**Sergio Torres**

**Pull request merged**